### PR TITLE
fix(replaceFieldOperator): disallow to pass null because it will emulate instead to remove the operator

### PR DESCRIPTION
**Pull request by Scra3**
…

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by Scra3** [comment]
Task linked: [CU-86bx3z5xw 🔥🔥 P2 - Agent replaceFieldOperator - remove the doc "disabling operator"](https://app.clickup.com/t/86bx3z5xw)

File: packages/datasource-customizer/src/decorators/operators-emulate/collection.ts [hunk 1]
```diff
@@ -21,13 +21,22 @@ import CollectionCustomizationContext from '../../context/collection-context';
...
-  private readonly fields: Map<string, Map<Operator, OperatorDefinition>> = new Map();
+  private readonly fields: Map<string, Map<Operator, OperatorDefinition | null>> = new Map();
 
   emulateFieldOperator(name: string, operator: Operator): void {
-    this.replaceFieldOperator(name, operator, null);
+    this.replaceOrEmulateFieldOperator(name, operator, null);
   }
 
   replaceFieldOperator(name: string, operator: Operator, replaceBy: OperatorDefinition): void {
+    if (!replaceBy) throw new Error('replace handler is required');
```
Comment (Thenkei) [inline packages/datasource-customizer/src/decorators/operators-emulate/collection.ts:31]:
```suggestion
    if (!replaceBy) throw new Error('replaceBy handler is required');
```

File: packages/datasource-customizer/test/decorators/operators-emulate/collection.test.ts [hunk 2]
```diff
@@ -157,6 +157,14 @@ describe('OperatorsEmulateCollectionDecorator', () => {
       });
     });
 
+    describe('when given null handler value for replaceFieldOperator', () => {
```
Comment (Thenkei) [inline packages/datasource-customizer/test/decorators/operators-emulate/collection.test.ts:160]:
```suggestion
     test('replaceFieldOperator() should throw if no handler is given', () => {
```
If we want to stick to the previous tests hierarchy.

**Review by Thenkei — APPROVED** [review]
- ✅&nbsp;Code review
- ✅&nbsp;Automatic tests
  - ✅&nbsp; Unit tests
- ✅&nbsp;Manual tests
- ✅&nbsp;PR title
- ✅&nbsp;PR linked to the clickup task

File: packages/datasource-customizer/test/decorators/operators-emulate/collection.test.ts [hunk 3]
```diff
@@ -157,6 +157,14 @@ describe('OperatorsEmulateCollectionDecorator', () => {
       });
     });
 
+    test('replaceFieldOperator() should throw if no handler is given', () => {
+      test('should crash', async () => {
```
Comment (Thenkei) [inline packages/datasource-customizer/test/decorators/operators-emulate/collection.test.ts:161]:
```suggestion
```

File: packages/datasource-customizer/test/decorators/operators-emulate/collection.test.ts [hunk 4]
```diff
@@ -157,6 +157,14 @@ describe('OperatorsEmulateCollectionDecorator', () => {
...
+        expect(() => newBooks.replaceFieldOperator('title', 'StartsWith', null)).toThrow(
+          'replaceBy handler is required',
+        );
+      });
```
Comment (Thenkei) [inline packages/datasource-customizer/test/decorators/operators-emulate/collection.test.ts:165]:
```suggestion
```

**Pull request merged**